packages/website/src/components/hooks/useHashState.ts

```diff
@@ -165,66 +172,33 @@ const writeStateToLocalStorage = (newState: ConfigModel): void => {
 function useHashState(
   initialState: ConfigModel,
 ): [ConfigModel, (cfg: Partial<ConfigModel>) => void] {
-  const [hash, setHash] = useState<string>(window.location.hash.slice(1));
   const [state, setState] = useState<ConfigModel>(() => ({
     ...initialState,
     ...retrieveStateFromLocalStorage(),
     ...parseStateFromUrl(window.location.hash.slice(1)),
   }));
-  const [tmpState, setTmpState] = useState<Partial<ConfigModel>>(() => ({
-    ...initialState,
-    ...retrieveStateFromLocalStorage(),
-    ...parseStateFromUrl(window.location.hash.slice(1)),
-  }));
 
-  useEffect(() => {
-    const newHash = window.location.hash.slice(1);
-    if (newHash !== hash) {
-      const newState = parseStateFromUrl(newHash);
-      if (newState) {
-        setState(newState);
-        setTmpState(newState);
-      }
-    }
-  }, [hash]);
+  const updateState = useCallback((cfg: Partial<ConfigModel>) => {
+    console.info('[State] updating config diff', cfg);
 
-  useEffect(() => {
-    const newState = { ...state, ...tmpState };
-    if (!shallowEqual(newState, state)) {
-      writeStateToLocalStorage(newState);
+    setState(oldState => {
+      const newState = { ...oldState, ...cfg };
       const newHash = writeStateToUrl(newState);
-      setState(newState);
-      setHash(newHash);
-
-      if (window.location.hash.slice(1) !== newHash) {
-        window.history.pushState(
-          undefined,
-          document.title,
-          `${window.location.pathname}#${newHash}`,
-        );
-      }
-    }
-  }, [tmpState, state]);
 
-  const onHashChange = (): void => {
-    const newHash = window.location.hash;
-    console.info('[State] hash change detected', newHash);
-    setHash(newHash);
-  };
-
-  useEffect(() => {
-    window.addEventListener('popstate', onHashChange);
-    return (): void => {
-      window.removeEventListener('popstate', onHashChange);
-    };
-  }, []);
+      writeStateToLocalStorage(newState);
+      const url = `${window.location.pathname}#${newHash}`;
 
-  const _setState = useCallback((cfg: Partial<ConfigModel>) => {
-    console.info('[State] updating config diff', cfg);
-    setTmpState(cfg);
+      if (cfg.ts) {
+        window.location.href = url;
+        window.location.reload();
+      } else {
+        window.history.replaceState(undefined, document.title, url);
```

Comment:
note: this should be changed to use react router to fix issues with navigation 

but this is oo scope for this change